db/migrations/20210517102049_create_user_community_table.ts

```diff
@@ -0,0 +1,19 @@
+import { Knex } from "knex";
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.schema.createTable("user_community", (table) => {
+    table.increments().primary();
+    table.integer("user_id").references("id").inTable("user").notNullable();
+    table
+      .integer("community_id")
+      .references("id")
+      .inTable("community")
+      .notNullable();
```

Comment:
Consider adding `.index()` for `community_id` and `user_id`.

Adding indexes on `*_id` columns of linking tables is quite valuable - it speeds up queries involving that table significantly.